File: Code/src/main/java/com/jmd/ui/common/CommonDialog.java
package com.jmd.ui.common;

import com.jmd.ApplicationStore;

import javax.swing.*;
import java.util.Arrays;

public class CommonDialog {

    public static void alert(String title, String message) {
        var pane = new JOptionPane();
        pane.setMessageType(JOptionPane.INFORMATION_MESSAGE);
        pane.setMessage(message);
        if (title == null) {
            title = UIManager.getString("OptionPane.messageDialogTitle", ApplicationStore.commonParentFrame.getLocale());
        }
        var dialog = pane.createDialog(ApplicationStore.commonParentFrame, title);
        var x = ApplicationStore.MAIN_FRAME_LOCATION_X - (int) dialog.getSize().getWidth() / 2 + ApplicationStore.MAIN_FRAME_WIDTH / 2;
        var y = ApplicationStore.MAIN_FRAME_LOCATION_Y - (int) dialog.getSize().getHeight() / 2 + ApplicationStore.MAIN_FRAME_HEIGHT / 2;
        dialog.setLocation(x, y);
        dialog.setVisible(true);
        dialog.dispose();
    }

    public static boolean confirm(String title, String message) {
        var pane = new JOptionPane();
        pane.setMessageType(JOptionPane.QUESTION_MESSAGE);
        pane.setMessage(message);
        pane.setOptionType(JOptionPane.YES_NO_OPTION);
        if (title == null) {
            title = UIManager.getString("OptionPane.messageDialogTitle", ApplicationStore.commonParentFrame.getLocale());
        }
        var dialog = pane.createDialog(ApplicationStore.commonParentFrame, title);
        pane.selectInitialValue();
        var x = ApplicationStore.MAIN_FRAME_LOCATION_X - (int) dialog.getSize().getWidth() / 2 + ApplicationStore.MAIN_FRAME_WIDTH / 2;
        var y = ApplicationStore.MAIN_FRAME_LOCATION_Y - (int) dialog.getSize().getHeight() / 2 + ApplicationStore.MAIN_FRAME_HEIGHT / 2;
        dialog.setLocation(x, y);
        dialog.setVisible(true);
        dialog.dispose();
        var selectedValue = pane.getValue();
        if (selectedValue == null) {
            return false;
        }
        if (selectedValue instanceof Integer) {
            return selectedValue.equals(JOptionPane.YES_OPTION);
        }
        return false;
    }

    public static Integer option(String title, String message, String[] options) {
        var pane = new JOptionPane();
        pane.setMessageType(JOptionPane.QUESTION_MESSAGE);
        pane.setMessage(message);
        pane.setOptions(Arrays.copyOf(options, options.length));
        if (title == null) {
            title = UIManager.getString("OptionPane.messageDialogTitle", ApplicationStore.commonParentFrame.getLocale());
        }
        var dialog = pane.createDialog(ApplicationStore.commonParentFrame, title);
        pane.selectInitialValue();
        var x = ApplicationStore.MAIN_FRAME_LOCATION_X - (int) dialog.getSize().getWidth() / 2 + ApplicationStore.MAIN_FRAME_WIDTH / 2;
        var y = ApplicationStore.MAIN_FRAME_LOCATION_Y - (int) dialog.getSize().getHeight() / 2 + ApplicationStore.MAIN_FRAME_HEIGHT / 2;
        dialog.setLocation(x, y);
        dialog.setVisible(true);
        dialog.dispose();
        var selectedValue = pane.getValue();
        if (selectedValue == null) {
            return null;
        }
        for (var i = 0; i < options.length; i++) {
            if (options[i].equals(selectedValue)) {
                return i;
            }
        }
        return null;
    }

}

File: Code/src/main/java/com/jmd/ui/common/CommonDialog.kt
package com.jmd.ui.common

import com.jmd.ApplicationStore
import java.util.*
import javax.swing.JOptionPane
import javax.swing.UIManager

object CommonDialog {

    @JvmStatic
    fun alert(title: String?, message: String?) {
        var titleValue = title
        val pane = JOptionPane()
        pane.messageType = JOptionPane.INFORMATION_MESSAGE
        pane.message = message
        if (titleValue == null) {
            titleValue = UIManager.getString("OptionPane.messageDialogTitle", ApplicationStore.commonParentFrame?.locale)
        }
        val dialog = pane.createDialog(ApplicationStore.commonParentFrame, titleValue)
        val lx = ApplicationStore.MAIN_FRAME_LOCATION_X
        val ly = ApplicationStore.MAIN_FRAME_LOCATION_Y
        val mw = ApplicationStore.MAIN_FRAME_WIDTH
        val mh = ApplicationStore.MAIN_FRAME_HEIGHT
        val x = lx - dialog.size.getWidth().toInt() / 2 + mw / 2
        val y = ly - dialog.size.getHeight().toInt() / 2 + mh / 2
        dialog.setLocation(x, y)
        dialog.isVisible = true
        dialog.dispose()
    }

    @JvmStatic
    fun confirm(title: String?, message: String?): Boolean {
        var titleValue = title
        val pane = JOptionPane()
        pane.messageType = JOptionPane.QUESTION_MESSAGE
        pane.message = message
        pane.optionType = JOptionPane.YES_NO_OPTION
        if (titleValue == null) {
            titleValue = UIManager.getString("OptionPane.messageDialogTitle", ApplicationStore.commonParentFrame?.locale)
        }
        val dialog = pane.createDialog(ApplicationStore.commonParentFrame, titleValue)
        pane.selectInitialValue()
        val lx = ApplicationStore.MAIN_FRAME_LOCATION_X
        val ly = ApplicationStore.MAIN_FRAME_LOCATION_Y
        val mw = ApplicationStore.MAIN_FRAME_WIDTH
        val mh = ApplicationStore.MAIN_FRAME_HEIGHT
        val x = lx - dialog.size.getWidth().toInt() / 2 + mw / 2
        val y = ly - dialog.size.getHeight().toInt() / 2 + mh / 2
        dialog.setLocation(x, y)
        dialog.isVisible = true
        dialog.dispose()
        val selectedValue = pane.value ?: return false
        return if (selectedValue is Int) {
            selectedValue == JOptionPane.YES_OPTION
        } else false
    }

    @JvmStatic
    fun option(title: String?, message: String?, options: Array<String>): Int? {
        var titleValue = title
        val pane = JOptionPane()
        pane.messageType = JOptionPane.QUESTION_MESSAGE
        pane.message = message
        pane.options = Arrays.copyOf(options, options.size)
        if (titleValue == null) {
            titleValue = UIManager.getString("OptionPane.messageDialogTitle", ApplicationStore.commonParentFrame?.locale)
        }
        val dialog = pane.createDialog(ApplicationStore.commonParentFrame, titleValue)
        pane.selectInitialValue()
        val lx = ApplicationStore.MAIN_FRAME_LOCATION_X
        val ly = ApplicationStore.MAIN_FRAME_LOCATION_Y
        val mw = ApplicationStore.MAIN_FRAME_WIDTH
        val mh = ApplicationStore.MAIN_FRAME_HEIGHT
        val x = lx - dialog.size.getWidth().toInt() / 2 + mw / 2
        val y = ly - dialog.size.getHeight().toInt() / 2 + mh / 2
        dialog.setLocation(x, y)
        dialog.isVisible = true
        dialog.dispose()
        val selectedValue = pane.value ?: return null
        for (i in options.indices) {
            if (options[i] == selectedValue) {
                return i
            }
        }
        return null
    }

}